app/controllers/application_controller.rb

```diff
@@ -49,6 +55,28 @@ def default_html_head
 
   protected
 
+  # A common method for controllers to find a Dor object.  It accepts any valid
+  # PID (druid:aa111aa1111) or DRUID (aa111aa1111).  DruidTools::Druid parses
+  # input and returns a uniform PID, which is input to Dor.find to obtain a Dor
+  # object.  If the object identifier is not a valid DRUID/PID, it will generate
+  # a UI alert and log a warning, but allow Dor.find to try to get it.
+  # @see https://github.com/sul-dlss/druid-tools
+  # @param druid [String]
+  # @return object [ActiveFedora::Base] anything returned by Dor.find
+  # @raises any exceptions from Dor.find
+  def find_druid(druid)
+    pid = DruidTools::Druid.new(druid).druid
+    Dor.find(pid)
+  rescue ArgumentError
+    # Allow Argo to find anything using Dor.find,
+    # in case there is bad IDs that need to be remediated.
+    # The Dor.find method should be read-only access.
+    msg = "Trying to find an invalid DRUID: #{druid}"
+    logger.warn msg
+    flash[:alert] = msg
+    Dor.find(druid)
+  end
+
```

Comment:
This method is now protected and it allows Dor.find requests for invalid DRUIDs in case they need to be remediated.